### PR TITLE
Improve pdfplumber parsing to format markdown and detect hyperlinks

### PR DESCRIPTION
**Pull request by dilithjay**
The `Rect` attributes of each URI in the PDF byte string are used to locate the text corresponding to the hidden URIs.

### CODE REVIEW

**Comment by pramitchoudhary** [comment]
One of benchmark, i was referring to earlier --> https://github.com/py-pdf/benchmarks

File: lexoid/core/parse_type/static_parser.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 import pymupdf4llm
+from typing import List, Dict
+from lexoid.core.utils import get_uri_rect, split_pdf
 from pdfminer.high_level import extract_pages
 from pdfminer.layout import LTTextContainer
-from pdfplumber.utils import extract_text, get_bbox_overlap, obj_to_bbox
+from pdfplumber.utils import get_bbox_overlap, obj_to_bbox
 
 
 def parse_static_doc(path: str, raw: bool, **kwargs) -> List[Dict] | str:
```
Comment (pramitchoudhary) [inline lexoid/core/parse_type/static_parser.py:12]: Should we change the default framework in `parse_static_doc`?

**Review by pramitchoudhary — APPROVED** [review]
Minor suggestion, otherwise LGTM. Thanks @dilithjay 🙏

File: lexoid/core/parse_type/static_parser.py (hunk 1 above)
Comment (dilithjay) [reply lexoid/core/parse_type/static_parser.py:12]: Done

**Pull request merged**